src/amo/reducers/collections.js

```diff
@@ -0,0 +1,160 @@
+/* @flow */
+import { createInternalAddon } from 'core/reducers/addons';
+import type { AddonType, ExternalAddonType } from 'core/types/addons';
+
+export const FETCH_COLLECTION = 'FETCH_COLLECTION';
+export const LOAD_COLLECTION = 'LOAD_COLLECTION';
+
+export type CollectionType = {
+  addons: Array<AddonType>,
+  authorName: string,
+  description: string | null,
+  id: number,
+  lastUpdatedDate: string,
+  name: string,
+  numberOfAddons: number,
+};
+
+export type CollectionsState = {|
+  current: CollectionType | null,
+  loading: boolean,
+|};
+
+export const initialState: CollectionsState = {
+  current: null,
+  loading: false,
+};
+
+type FetchCollectionParams = {|
+  errorHandlerId: string,
+  page?: number,
+  slug: string,
+  user: number | string,
+|};
+
+export const fetchCollection = ({
+  errorHandlerId,
+  page,
+  slug,
+  user,
+}: FetchCollectionParams = {}) => {
+  if (!errorHandlerId) {
+    throw new Error('errorHandlerId is required');
+  }
+  if (!slug) {
+    throw new Error('slug is required');
+  }
+  if (!user) {
+    throw new Error('user is required');
+  }
+
+  return {
+    type: FETCH_COLLECTION,
+    payload: { errorHandlerId, page, slug, user },
+  };
+};
+
+type CollectionAddonsResults = Array<{|
+  addon: ExternalAddonType,
+  downloads: number,
+  notes: string | null,
+|}>;
+
+// The API returns more attributes than what is listed below.
+type CollectionDetail = {
```

Comment:
If I'm understanding correctly, this is the API response that eventually becomes `CollectionType`, correct? If so, you should name this one `ExternalCollectionType` to make it clear.

Reply:
Not exactly, the `CollectionType` is the representation I need in the app, but it merges different result sets because there are 2 endpoints to call.